### PR TITLE
BT 1210 BUIL 3685

### PR DESCRIPTION
**Pull request by HarshGandhi197**
Change Order RollUp trigger field methods added on project

### CODE REVIEW

File: force-app/main/default/classes/ChangeOrderTriggerHandler.cls [hunk 1]
```diff
@@ -1124,4 +1124,146 @@ public without sharing class ChangeOrderTriggerHandler {
...
+            List<buildertek__Change_Order__c> COList = new List<buildertek__Change_Order__c>();
+
+            for(buildertek__Change_Order__c co : oldChangeOrder){
+                if(co.buildertek__Status__c != 'Pending' && co.buildertek__Status__c != 'Company Rejected' && co.buildertek__Status__c != 'Customer Rejected'){
+                prId.add(co.buildertek__Project__c);
+                }
+            }
+            System.debug('prId ==> ' + prId);
+
+            COList = [SELECT Id,buildertek__Project__c,buildertek__Total_Amount_Tax__c FROM buildertek__Change_Order__c WHERE buildertek__Project__c IN :prId AND buildertek__Status__c NOT IN ('Pending','Company Rejected','Customer Rejected')];
```
Comment (Nishitsuthar) [inline force-app/main/default/classes/ChangeOrderTriggerHandler.cls:1244]: again no need to check status in query

**Review by Nishitsuthar — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/ChangeOrderTriggerHandler.cls [hunk 2]
```diff
@@ -1124,4 +1124,146 @@ public without sharing class ChangeOrderTriggerHandler {
...
+            List<buildertek__Change_Order__c> COList = new List<buildertek__Change_Order__c>();
+
+            for(buildertek__Change_Order__c co : newChangeOrder){
+                if(co.buildertek__Status__c != 'Pending' && co.buildertek__Status__c != 'Company Rejected' && co.buildertek__Status__c != 'Customer Rejected' && co.buildertek__Project__c != null){
+                    prId.add(co.buildertek__Project__c);
+                }
+            }
+            System.debug('prid ==> ' + prId);
+
+            COList = [SELECT Id,buildertek__Project__c,buildertek__Total_Amount_Tax__c FROM buildertek__Change_Order__c WHERE buildertek__Project__c IN :prId AND buildertek__Status__c NOT IN ('Pending','Company Rejected','Customer Rejected')];
```
[the remark this answers is not in the document]
Comment (HarshGandhi197) [reply force-app/main/default/classes/ChangeOrderTriggerHandler.cls]: Changes Done

File: force-app/main/default/classes/ChangeOrderTriggerHandler.cls [hunk 3]
```diff
@@ -1124,4 +1124,146 @@ public without sharing class ChangeOrderTriggerHandler {
...
+                bpr.buildertek__Total_Change_Orders__c = 0;
+                for(buildertek__Change_Order__c bco : oldCOList){
+                    bpr.buildertek__Total_Change_Orders__c += bco.buildertek__Total_Amount_Tax__c;
+                }
+                projectList.add(bpr);
+            }
+
+            // to update new
+            for(buildertek__Change_Order__c co : updatedChangeOrder){
+                // if(co.buildertek__Status__c != 'Pending' && co.buildertek__Status__c != 'Company Rejected' && co.buildertek__Status__c != 'Customer Rejected' && co.buildertek__Project__c != null){
```
[the remark this answers is not in the document]
Comment (HarshGandhi197) [reply force-app/main/default/classes/ChangeOrderTriggerHandler.cls]: Removed unnecessary code

File: force-app/main/default/classes/ChangeOrderTriggerHandler.cls (hunk 1 above)
Comment (HarshGandhi197) [reply force-app/main/default/classes/ChangeOrderTriggerHandler.cls]: Changed done

**Review by Nishitsuthar — APPROVED** [review]

**Pull request merged**